### PR TITLE
Graduate HugePages feature to GA

### PR DESCRIPTION
**Pull request by derekwaynecarr**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Graduates the feature that has been beta for 2 releases.
KEP https://github.com/kubernetes/enhancements/pull/750

**Special notes for your reviewer**:
Pending sig-node discussion.

**Does this PR introduce a user-facing change?**:
```release-note
The HugePages feature gate has graduated to GA, and can no longer be disabled. The feature gate will be removed in v1.16
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -410,7 +411,7 @@ var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureS
 	RotateKubeletClientCertificate:              {Default: true, PreRelease: utilfeature.Beta},
 	PersistentLocalVolumes:                      {Default: true, PreRelease: utilfeature.Beta},
 	LocalStorageCapacityIsolation:               {Default: true, PreRelease: utilfeature.Beta},
-	HugePages:                                   {Default: true, PreRelease: utilfeature.Beta},
+	HugePages:                                   {Default: true, PreRelease: utilfeature.GA},
```
Comment (liggitt) [inline pkg/features/kube_features.go]: recommendation is to also set `, LockToDefault: true` here and remove checking of the feature gate everywhere

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -400,8 +401,8 @@ func init() {
 // To add a new feature, define a key for it above and add it here. The features will be
 // available throughout Kubernetes binaries.
 var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureSpec{
-	AppArmor:             {Default: true, PreRelease: utilfeature.Beta},
-	DynamicKubeletConfig: {Default: true, PreRelease: utilfeature.Beta},
+	AppArmor:                                    {Default: true, PreRelease: utilfeature.Beta},
+	DynamicKubeletConfig:                        {Default: true, PreRelease: utilfeature.Beta},
```
Comment (liggitt) [inline pkg/features/kube_features.go]: verify gofmt was unhappy with this change, might be your local go version

**Comment by liggitt** [comment]
lgtm modulo test failure

**Comment by dchen1107** [comment]
Please resolve the rebase issue; otherwise it is ready to go.

/lgtm
/approve

**Pull request merged**